erpnext/stock/doctype/item/item.py

```diff
@@ -30,9 +30,16 @@ def onload(self):
 		self.get("__onload").sle_exists = self.check_if_sle_exists()
 
 	def autoname(self):
-		if frappe.db.get_default("item_naming_by")=="Naming Series" and not self.variant_of:
-			from frappe.model.naming import make_autoname
-			self.item_code = make_autoname(self.naming_series+'.#####')
+		if frappe.db.get_default("item_naming_by")=="Naming Series":
+			if self.variant_of:
+				item_code_suffix = ""
```

Comment:
Item code for variants can be changed manually. Hence, it should set auto_name **only if item_code is blank**. Also, when creating variant using UI, item_code is already set.